c++/src/capnp/compat/byte-stream.c++

```diff
@@ -240,6 +245,30 @@ public:
     KJ_UNREACHABLE;
   }
 
+  kj::Promise<void> startTls(StartTlsContext context) override {
```

Comment:
A substream allowing TLS initiation strikes me as a little weird, because from reading the code, it seems that substreams' primary use case is to optimize write operations to the parent stream. Presumably those write operations will always have completed before TLS initiation.

Is this code path exercised anywhere, do you know? If not, and if it's hard to test, and if we can't think of a case where we'd need it, maybe it should unconditionally throw?

Reply:
Yeah, I think we can `throw` here.